packages/legend-studio-extension-query-builder/src/components/QueryBuilderDialog.tsx

```diff
@@ -42,13 +46,37 @@ export const QueryBuilderDialog = observer(() => {
   );
   const [isMaximized, setIsMaximized] = useState(false);
   const toggleMaximize = (): void => setIsMaximized(!isMaximized);
+  const USER_ATTESTATION_MESSAGE =
+    'Unsaved changes to your query will be lost if you continue. Are you sure?';
+
   const closeQueryBuilder = (): void => {
     flowResult(
       queryBuilderExtensionState.setEmbeddedQueryBuilderMode(undefined),
     ).catch(applicationStore.alertUnhandledError);
     queryBuilderExtensionState.reset();
   };
 
+  const confirmCloseQueryBuilder = (): void => {
```

Comment:
@MauricioUyaguari with this, do we want to do change detection for the query?

Reply:
For this I think we would only warn when they query is coming from a persistent state. 
I.e service editor, mapping editor, saved query. 
We should calculate if there has been any changes then ask this if there has been.

Reply:
I monitored the query lambda. If there are any changes in lambda, it means that users make some changes.